packages/template-retail-react-app/app/ssr.js

```diff
@@ -35,19 +35,53 @@ const options = {
 const runtime = getRuntime()
 
 const {handler} = runtime.createHandler(options, (app) => {
+    const getCSP = (nodeEnv, site) => {
+        const trustedMap = {
+            development: [
+                'localhost:*',
+                '*.mobify-staging.com',
+                '*.demandware.net',
+                '*.mobify-storefront-staging.com',
+                '*.mobify-storefront.com',
+                '*.commercecloud.salesforce.com',
+                'runtime.commercecloud.com'
+            ],
+            staging: [
+                '*.mobify-staging.com',
+                '*.demandware.net',
+                '*.mobify-storefront-staging.com',
+                '*.mobify-storefront.com',
+                '*.commercecloud.salesforce.com',
+                'runtime.commercecloud.com'
+            ],
+            production: [
+                '*.mobify.com',
+                '*.demandware.com',
+                '*.mobify-storefront.com',
+                '*.commercecloud.salesforce.com',
+                'runtime.commercecloud.com'
+            ]
+        }
+        const trusted = ["'self'", ...(trustedMap[nodeEnv] ? trustedMap[nodeEnv] : [])]
+        return {
+            directives: {
+                'connect-src': ['api.cquotient.com', ...trusted],
+                'frame-ancestors': [...trusted],
+                'img-src': ['data:', ...trusted],
+                'script-src': ['unsafe-eval', 'storage.googleapis.com', ...trusted],
+
+                // Do not upgrade insecure requests for local development
+                'upgrade-insecure-requests': isRemote() ? [] : null
+            }
```

Comment:
```suggestion
                'connect-src': ['api.cquotient.com', ...trusted],
                'frame-ancestors': [...trusted],
                'img-src': ['data:', ...trusted],
                'script-src': ['unsafe-eval', 'storage.googleapis.com', ...trusted],

                // Do not upgrade insecure requests for local development
                'upgrade-insecure-requests': isRemote() ? [] : null
```
This should just return the directives, not the wrapped object, for the code to work. 🙂